core/src/snapshot_utils.rs

```diff
@@ -105,65 +170,40 @@ pub fn remove_snapshot<P: AsRef<Path>>(slot: u64, snapshot_path: P) -> Result<()
     Ok(())
 }
 
-pub fn bank_from_snapshots<P, Q>(
+pub fn bank_from_snapshots<P>(
     local_account_paths: String,
-    snapshot_path: P,
-    append_vecs_path: Q,
+    snapshot_paths: &[SlotSnapshotPaths],
+    append_vecs_path: P,
 ) -> Result<Bank>
 where
     P: AsRef<Path>,
-    Q: AsRef<Path>,
 {
     // Rebuild the last root bank
-    let names = get_snapshot_names(&snapshot_path);
-    let last_root = names
+    let last_root_paths = snapshot_paths
         .last()
         .ok_or_else(|| get_io_error("No snapshots found in snapshots directory"))?;
-    let snapshot_file_name = get_snapshot_file_name(*last_root);
-    let snapshot_dir = get_bank_snapshot_dir(&snapshot_path, *last_root);
-    let snapshot_file_path = snapshot_dir.join(&snapshot_file_name);
-    info!("Load from {:?}", snapshot_file_path);
-    let file = File::open(snapshot_file_path)?;
+    info!("Load from {:?}", &last_root_paths.snapshot_file_path);
+    let file = File::open(&last_root_paths.snapshot_file_path)?;
     let mut stream = BufReader::new(file);
     let bank: Bank = deserialize_from(&mut stream).map_err(|e| get_io_error(&e.to_string()))?;
 
     // Rebuild accounts
     bank.rc
         .accounts_from_stream(&mut stream, local_account_paths, append_vecs_path)?;
 
-    for bank_slot in names.iter().rev() {
-        let snapshot_file_name = get_snapshot_file_name(*bank_slot);
-        let snapshot_dir = get_bank_snapshot_dir(&snapshot_path, *bank_slot);
-        let snapshot_file_path = snapshot_dir.join(snapshot_file_name.clone());
-        let file = File::open(snapshot_file_path)?;
-        let mut stream = BufReader::new(file);
-        let _bank: Result<Bank> =
-            deserialize_from(&mut stream).map_err(|e| get_io_error(&e.to_string()));
-
-        // TODO: Uncomment and deserialize status cache here
-
-        /*let status_cache: Result<StatusCacheRc> = deserialize_from(&mut stream)
-        .map_err(|e| get_io_error(&e.to_string()));
-        if bank_root.is_some() {
-            match bank {
-                Ok(v) => {
-                    if status_cache.is_ok() {
-                        let status_cache = status_cache?;
-                        status_cache_rc.append(&status_cache);
-                        // On the last snapshot, purge all outdated status cache
-                        // entries
-                        if i == names.len() - 1 {
-                            status_cache_rc.purge_roots();
-                        }
-                    }
-
-                    bank_maps.push((*bank_slot, parent_slot, v));
-                }
-                Err(_) => warn!("Load snapshot failed for {}", bank_slot),
-            }
-        } else {
-            warn!("Load snapshot rc failed for {}", bank_slot);
-        }*/
+    // merge the status caches from all previous banks
+    for (i, slot_paths) in snapshot_paths.iter().rev().enumerate() {
```

Comment:
Is there a test, something like `test_bank_forks_snapshot_n` that tests this code path? Also, we should re-enable the commented part of the test  function `compare_bank` in `bank.rs`